src/variables/components/TypeAheadVariableDropdown.tsx

```diff
@@ -51,107 +56,184 @@ class TypeAheadVariableDropdown extends PureComponent<Props, MyState> {
+
+    const justLoaded = () =>
+      status === RemoteDataState.Done && prevStatus !== RemoteDataState.Done
+
+    const justSelected = () =>
+      selectHappened &&
+      !prevSelectHappened &&
+      actualVal &&
+      actualVal !== prevActualVal
+
     // this is for updating the values:
     // (only want this to run *once* when the values get loaded)
-    if (!loaded && prevVals.length !== values.length) {
-      this.setState({
+    if (justLoaded()) {
+      newState = this.getInitialValuesAfterLoading()
+    } else if (
+      !loaded &&
+      status === RemoteDataState.Done &&
+      prevVals.length !== values.length
+    ) {
+      newState = {
         shownValues: values,
         typedValue: selectedValue,
         loaded: true,
-      })
+      }
     }
 
-    // unset the menuOpen; it should be set to closed (or open) only once; then undone
+    /**
+     * unset the menuOpen; it should be set to closed (or open) only once; then undone
+     * this is needed because: (from clockface Dropdown.tsx documentation):
+     * if the string is set to 'open', and then the user closes it, and the code sets it to open again,
+     * unless the code sets it to something else in between (like null or 'close'),
+     * then nothing will happen- the menu will not open)
+     */
+
     if (menuOpen !== prevMenuOpen && menuOpen !== null) {
-      this.setState({menuOpen: null})
+      newState['menuOpen'] = null
     }
 
     // need to have this, as the 'onClickAwayHere' gets triggered *before*
     // the selected value is set to the actualValue (it keeps re-using the original
     // property)
 
     // for updating the selected value:
-    if (
-      selectHappened &&
-      !prevSelectHappened &&
-      actualVal &&
-      actualVal !== prevActualVal
-    ) {
-      this.setState({typedValue: actualVal, selectHappened: false})
+    if (justSelected()) {
+      newState = {...newState, typedValue: actualVal, selectHappened: false}
+    }
+
+    this.setState(newState)
+  }
+
+  getInitialValuesAfterLoading() {
+    const {values, selectedValue} = this.props
+    // it reloaded; maybe because of a dependent var
+    // want to re-init
+
+    // if selectedValue is present in the values, set it; else zero it out
+    let newSelectedValue = ''
+    if (values.includes(selectedValue)) {
+      newSelectedValue = selectedValue
+    }
+
+    return {
+      shownValues: values,
+      typedValue: newSelectedValue,
+      loaded: true,
+      actualVal: newSelectedValue,
     }
   }
 
-  filterVals = needle => {
+  filterVals = event => {
     const {values} = this.props
 
+    const needle = event?.target?.value
+
+    // if there is no value, set the shownValues to everything
+    // and set the typedValue to nothing (zero it out)
+    // reset the selectIndex too
     if (!needle) {
-      this.setState({shownValues: values, typedValue: needle})
+      this.setState({shownValues: values, typedValue: needle, selectIndex: -1})
     } else {
```

Comment:
i know we already shipped this code, but stylistically, i think this `if` statement should return early. the `else` statement is actually the function body that does what the function is named, but it's easy to miss because it's part of an `else` which has let weight than an `if` statement. if the `if` was transformed into a guard statement that returned early after it set state, the filtering could just be on equal indentation as the `if` statement